common/lockmanager_test.go

```diff
@@ -0,0 +1,318 @@
+package common
+
+import (
+    "sync"
+    "testing"
+    "time"
+
+    "github.com/edulinq/autograder/config"
+)
+
+func TestLockBase(test *testing.T) {
+    key1 := "testkey1";
+
+    Lock(key1);
+
+    err := Unlock(key1);
+    if (err != nil) {
+        test.Fatalf("Failed to unlock.");
+    }
+}
+
+func TestUnlockingAnUnlockedLock(test *testing.T) {
+    key1 := "testkey1";
+
+    Lock(key1);
+
+    err := Unlock(key1);
+    if (err != nil) {
+        test.Fatalf("Failed to unlock.");
+    }
+
+    err = Unlock(key1);
+    if (err == nil) {
+        test.Fatalf("Failed to return an error when unlocking a lock that is already unlocked.");
+    }
+}
+
+func TestUnlockingKeyThatDoesntExist(test *testing.T) {
+    doesNotExistKey := "dne";
+
+    err := Unlock(doesNotExistKey);
+    if (err == nil) {
+        test.Fatalf("Failed to return an error when unlocking a key that does not exist.");
+    }
+}
+
+func TestLockUnlockDifferentKeysOneThread(test *testing.T) {
```

Comment:
Should the function-level comment replace the individual-lined comments, and if so, should all the tests have a function-leveled comment for consistency instead of individual-lined comments?

Reply:
If the function-level comment is enough to make the test clear, then it can replace the individual comments.
I won't know until I see it, so it's up to you to see if they would be sufficient for someone reading the tests.